packages/pipeline-editor/src/NotebookSubmissionDialog.tsx

```diff
@@ -0,0 +1,138 @@
+ */
+
+import { IDictionary } from '@elyra/application';
+import * as React from 'react';
+
+import { IRuntime } from './PipelineService';
+import Utils from './utils';
+
+interface IProps {
+  runtimes: IRuntime[];
+  images: IDictionary<string>;
+  env: string[];
+}
+
+const EnvForm = ({ env }: { env: string[] }): JSX.Element => {
+  if (env.length > 0) {
+    return (
+      <>
+        <tr>
+          <td colSpan={4}></td>
+        </tr>
+        <tr>
+          <td colSpan={4}>
+            <div style={{ fontSize: 'var(--jp-ui-font-size3)' }}>
+              Environmental Variables
+            </div>
+          </td>
+        </tr>
+        {Utils.chunkArray(env, 4).map((col, i) => (
+          <tr key={i}>
+            {col.map(envVar => (
+              <td key={envVar}>
+                <label htmlFor={envVar}>{envVar}:</label>
+                <br />
+                <input
+                  type="text"
+                  id={envVar}
+                  className="envVar"
+                  name={envVar}
+                  size={20}
+                />
+              </td>
+            ))}
+          </tr>
+        ))}
+      </>
+    );
+  }
+  return null;
+};
+
+class NotebookSubmissionDialog extends React.Component<IProps> {
+  render(): React.ReactNode {
+    const { runtimes, images, env } = this.props;
+    return (
+      <form>
+        <table id="table-submit-dialog" className="elyra-table">
+          <tbody>
+            <tr>
+              <td colSpan={2}>
+                <label htmlFor="runtime_config">Runtime Config:</label>
+                <br />
+                <select
+                  id="runtime_config"
+                  name="runtime_config"
+                  className="elyra-form-runtime-config"
+                >
+                  {runtimes.map(runtime => (
+                    <option key={runtime.name} value={runtime.name}>
+                      {runtime.display_name}
+                    </option>
+                  ))}
+                </select>
+              </td>
+              <td colSpan={2}>
+                <label htmlFor="framework">Runtime images:</label>
+                <br />
+                <select
+                  id="framework"
+                  name="framework"
+                  className="elyra-form-framework"
+                >
+                  {Object.entries(images).map(([key, val]) => (
+                    <option key={key} value={key}>
+                      {val}
+                    </option>
+                  ))}
+                </select>
+              </td>
+            </tr>
+            <tr>
+              <td>
+                <br />
+                <input
+                  type="checkbox"
+                  id="dependency_include"
+                  name="dependency_include"
+                  size={20}
+                  defaultChecked
+                />
+                <label htmlFor="dependency_include">Include dependencies</label>
```

Comment:
missed this comment in my last review (even though I referenced it)

Previously this had a space before `Include dependencies`, this is the only visible change in this PR but I believe it is better this way without the space (makes it consistent with other checkboxes in forms)

Reply:
Yea, I removed the space to be consistent with the label in the other dialog. Should I leave it without the space?

Reply:
yeah it's better this way